pennylane/ops/qubit/parametric_ops.py

```diff
@@ -524,6 +549,18 @@ def compute_matrix(phi):  # pylint: disable=arguments-differ
             phi = qml.math.cast_like(phi, 1j)
 
         exp_part = qml.math.exp(1j * phi)
+        shape = qml.math.shape(phi)
+        if len(shape) > 0:
```

Comment:
```suggestion
        shape = qml.math.shape(phi)

        if len(shape) > 0:
```

[should this be checking ndim?]